From cce75c6fc5500482cd7376b3a521386defb0d6c6 Mon Sep 17 00:00:00 2001
From: Haihao Xiang <haihao.xiang@intel.com>
Date: Thu, 30 Jun 2022 15:05:31 +0800
Subject: [PATCH 49/51] lavc/qsvenc_{hevc, vp9}: 0yuv, y410 in system memory

$ ffmpeg -y -f rawvideo -pix_fmt 0yuv -s:v 352x288 -i in.yuv -c:v hevc_qsv out.h265
$ mediainfo out.h265
[...]
Chroma subsampling                       : 4:4:4
Bit depth                                : 8 bits
Color range                              : Limited
---
 libavcodec/qsvenc_hevc.c | 1 +
 libavcodec/qsvenc_vp9.c  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/libavcodec/qsvenc_hevc.c b/libavcodec/qsvenc_hevc.c
index 36fa4a397f..002edd31c3 100644
--- a/libavcodec/qsvenc_hevc.c
+++ b/libavcodec/qsvenc_hevc.c
@@ -321,6 +321,7 @@ const FFCodec ff_hevc_qsv_encoder = {
                                                     AV_PIX_FMT_BGRA,
                                                     AV_PIX_FMT_X2RGB10,
                                                     AV_PIX_FMT_VUYX,
+                                                    AV_PIX_FMT_XV30,
                                                     AV_PIX_FMT_NONE },
     .p.priv_class   = &class,
     .defaults       = qsv_enc_defaults,
diff --git a/libavcodec/qsvenc_vp9.c b/libavcodec/qsvenc_vp9.c
index c3d09f880b..3295ffb2ef 100644
--- a/libavcodec/qsvenc_vp9.c
+++ b/libavcodec/qsvenc_vp9.c
@@ -117,6 +117,7 @@ const FFCodec ff_vp9_qsv_encoder = {
                                                     AV_PIX_FMT_QSV,
                                                     AV_PIX_FMT_VAAPI,
                                                     AV_PIX_FMT_D3D11,
+                                                    AV_PIX_FMT_XV30,
                                                     AV_PIX_FMT_NONE },
     .p.priv_class   = &class,
     .defaults       = qsv_enc_defaults,
-- 
2.25.1

